core/src/main/java/org/elasticsearch/search/rescore/RescoreBuilder.java

```diff
@@ -78,23 +139,41 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
         protected abstract XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws IOException;
 
+        @Override
+        public abstract int hashCode();
+
+        @Override
+        public abstract boolean equals(Object obj);
     }
 
     public static class QueryRescorer extends Rescorer {
+
         private static final String NAME = "query";
-        private QueryBuilder queryBuilder;
-        private Float rescoreQueryWeight;
-        private Float queryWeight;
-        private String scoreMode;
+        public static final QueryRescorer PROTOTYPE = new QueryRescorer(new MatchAllQueryBuilder());
+        public static final float DEFAULT_RESCORE_QUERYWEIGHT = 1.0f;
+        public static final float DEFAULT_QUERYWEIGHT = 1.0f;
+        public static final QueryRescoreMode DEFAULT_SCORE_MODE = QueryRescoreMode.Total;
+        private QueryBuilder<?> queryBuilder;
```

Comment:
Talked with Simon and decided so solve this by letting ObjectParser parse to an intermediate structure that then emits the builder, so I can make this final.